android/V8Runtime_shared.cpp

```diff
@@ -812,6 +812,54 @@ namespace facebook { namespace v8runtime {
     return handle_scope.Escape(v8BigIntValue->v8BigInt_.Get(v8::Isolate::GetCurrent()));
   }
 
+  jsi::BigInt V8Runtime::createBigIntFromInt64(int64_t val) {
+    return make<jsi::BigInt>(makeBigIntValue(v8::BigInt::New(GetIsolate(), val)));
+  }
+
+  jsi::BigInt V8Runtime::createBigIntFromUint64(uint64_t val) {
+    return make<jsi::BigInt>(makeBigIntValue(v8::BigInt::NewFromUnsigned(GetIsolate(), val)));
+  }
+
+  bool V8Runtime::bigintIsInt64(const jsi::BigInt&) {
+    // V8 doesn't internally track it
+    return true;
+  }
+
+  bool V8Runtime::bigintIsUint64(const jsi::BigInt& val) {
+    bool lossless {true};
+    bigIntRef(val)->Uint64Value(&lossless);
+    return lossless;
+  }
+
+  uint64_t V8Runtime::truncate(const jsi::BigInt& val) {
+    return bigIntRef(val)->Uint64Value(nullptr);
+  }
+
+  jsi::String V8Runtime::bigintToString(const jsi::BigInt&, int) {
+    throw jsi::JSINativeException("V8Runtime::bigintToString is not implemented!");
+  }
+
+  bool V8Runtime::hasNativeState(const jsi::Object& obj) {
+    return objectRef(obj)->InternalFieldCount() == 1;
+  }
+
+  std::shared_ptr<jsi::NativeState> V8Runtime::getNativeState(const jsi::Object& obj) {
+    std::shared_ptr<jsi::NativeState>* holder = static_cast<std::shared_ptr<jsi::NativeState>*>(objectRef(obj)->GetAlignedPointerFromInternalField(0));
```

Comment:
>GetAlignedPointerFromInternalField(0 [](http://example.com/codeflow?start=112&length=36)

Is this guaranteed to return 0/NULL if native state is not set ?

Reply:
https://v8docs.nodesource.com/node-0.12/db/d85/classv8_1_1_object.html#a580ea84afb26c005d6762eeb9e3c308f sounds like if it isn't set, it is undefined behavior. I don't know how we would know if it was set.

Reply:
we will definitely need to revisit the 0.71 implementation... for now I'm just getting it compiling to block the rn bump.